libtcmu-generic.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017, Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+*/
+
+#include <stdlib.h>
+#include <errno.h>
+#include <scsi/scsi.h>
+
+#include "scsi_defs.h"
+#include "libtcmu_log.h"
+#include "libtcmu-generic.h"
+
+int generic_compare_and_write(struct tcmu_device *dev,
```

Comment:
Hey,

We have been putting emulated commands in api.c with names like tmcu_emulate_$command_name(). I think you would want to put this command there too.